### ISSUE TITLE
Makefile repeats various flags; is fragile

### ISSUE THREAD

**Issue by bboreham** [weaveworks/weave#399]
For instance `-tags netgo`; these could be wrapped up inside variables to ensure they are applied consistently.

`-cover` for tests is similar, post #396

**Comment by rade** [comment]
I don't think avoiding the repetition of `cover` is worth the hassle.

Ideally we'd be able to write

    $(GO) ...

instead of

    go ...

but the tags must come after the command, so making `$(GO)` expand to `go -tags netgo` doesn't work.

We could go the custom function route, but that makes the call sites look horrific.

So the best solution @dpw and i can think of is to introduce a

    GOFLAGS=-tags netgo

and then change all occurrences of

    go <cmd> ...

to

    go <cmd> $(GOFLAGS) ...

But this only ensures consistency if you pay attention and don't introduce 'naked' `go` invocations.

**Comment by rade** [comment]
I have just thought of another possibility...

Introduce a `netgo` helper script that invokes go with `-tags netgo` inserted after the first param.

The Makefile would then define `GO=netgo` and we'd invoke go everywhere as `$(GO) ...`.

**Comment by bboreham** [comment]
That reminds me: I think we should use -installsuffix along with -tags netgo. It puts the resulting files in different directories so you don't have to recompile when you switch.

**Comment by rade** [comment]
> I think we should use -installsuffix

[Apparently](https://github.com/golang/go/issues/10011) `-installsuffix` doesn't work particularly well when using a distro-provided go.

**Comment by rade** [comment]
This has [just](https://github.com/rade/weave/commit/338d577fb75a78cfc0506d7c69ecb239ec9aed95#diff-b67911656ef5d18c4ae36cb6741b7965R47) gotten more complicated :(

**Comment by rade** [comment]
I struggle to see how this would ever be worthwhile resolving, at least not until our Makefile grows substantially. Too much effort for too little gain. -> wontfix
